source/documentation/upgrade_guide/topics/installing_virtual_product_hypervisors_from_local_repository.adoc

```diff
@@ -0,0 +1,117 @@
+
+.Procedure
+
+. On the system hosting the offline repository, create a file named `/etc/yum.repos.d/rhvh-mirror.repo` with contents similar to the lines that follow:
++
+[options="nowrap" subs="normal"]
+----
+[rhvh-4-for-rhel-8-x86_64-rpms]
+name = Red Hat Virtualization Host for RHEL 8 x86_64 (RPMs)
+baseurl = https://cdn.redhat.com/content/dist/layered/rhel8/x86_64/rhvh/4/os
+enabled = 0
+gpgcheck = 1
+gpgkey = file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release
+sslverify = 1
+sslcacert = /etc/rhsm/ca/redhat-uep.pem
+metadata_expire = 86400
+enabled_metadata = 1
+sslclientcert =
+sslclientkey =
+----
++
+You must populate the `sslclientcert` and `sslclientkey` fields with full path names to the correct files containing the appropriate certificate and key. The `/etc/pki/entitlement` directory contains one or more pairs of certificate and key files, but only one pair contains the RHV-H entitlement you need.
+
+. To find the certificate file:
+
+.. List all files in the `/etc/pki/entitlement` directory:
++
+[options="nowrap" subs="normal"]
+----
+# cd /etc/pki/entitlement/
+# ls -al
+----
++
+Output similar to the following is displayed:
++
+[options="nowrap" subs="normal"]
+----
+total 836
+drwxr-xr-x.  2 root root    202 May 28 15:18 .
+drwxr-xr-x. 15 root root    208 Apr 23  2020 ..
+-rw-r--r--.  1 root root   3243 May 28 15:18 4522783034260408538-key.pem
+-rw-r--r--.  1 root root 152622 May 28 15:18 4522783034260408538.pem
+-rw-r--r--.  1 root root   3243 May 28 15:18 5659494963772844103-key.pem
+-rw-r--r--.  1 root root 343394 May 28 15:19 5659494963772844103.pem
+-rw-r--r--.  1 root root   3243 May 23 13:19 645832581386032208-key.pem
+-rw-r--r--.  1 root root 343389 May 23 13:19 645832581386032208.pem
+#
+----
+
+.. Use the `rct cat-cert` command on each certificate in order to find the one that contains the RHV-H entitlement:
++
+[options="nowrap" subs="normal"]
+----
+# cd /etc/pki/entitlement/
+# rct cat-cert 5659494963772844103.pem | grep rhvh/4/ | grep URL
+----
++
+Output similar to the following is displayed:
++
+[options="nowrap" subs="normal"]
+----
+        URL: /content/beta/rhel/server/7/$basearch/rhvh/4/os
+        URL: /content/dist/rhel/server/7/7Server/$basearch/rhvh/4/os
+        URL: /content/beta/layered/rhel8/x86_64/rhvh/4/os
+        URL: /content/dist/layered/rhel8/x86_64/rhvh/4/os
+----
+
+. Identify the correct certificate and fill in the `sslclientcert` and `sslclientkey` values in the previously mentioned `.repo` file:
++
+[options="nowrap" subs="normal"]
+----
+sslclientcert = /etc/pki/entitlement/5659494963772844103.pem
+sslclientkey = /etc/pki/entitlement/5659494963772844103-key.pem
+----
+
+. Run the `reposync` command in the appropriate directory:
++
+.. Use the 'pwd' command to determine the correct path:
++
+[options="nowrap" subs="normal"]
+----
+# pwd
+----
++
+Output similar to the following is displayed:
++
+[options="nowrap" subs="normal"]
+----
+# /home/test/rhvh-reposync
```

Comment:
this # is not what will be in output, there will be only directory path in output